src/client.ts

```diff
@@ -0,0 +1,162 @@
+import type { JsonObject, JsonValue } from "cockpit";
+
+import type { Connection, MonitorCallback } from "./rest.ts";
+
+// podman API version; oldest one that we support
+export const VERSION = "/v3.4.0/";
+
+const podmanCall = (con: Connection, name: string, method: string, args: JsonObject, body?: string):
+                   Promise<string> =>
+    con.call({ method, path: VERSION + name, body: body || "", params: args, });
+
+const podmanJson = (con: Connection, name: string, method: string, args: JsonObject, body?: string):
+                   Promise<JsonObject|JsonValue> =>
+    podmanCall(con, name, method, args, body)
+            .then(reply => JSON.parse(reply));
+
+export const streamEvents = (con: Connection, callback: MonitorCallback) =>
+    con.monitor(VERSION + "libpod/events", callback);
+
+export function getInfo(con: Connection): Promise<JsonObject> {
+    return new Promise((resolve, reject) => {
+        const timeout = setTimeout(() => reject(new Error("timeout")), 10000);
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2214-e5ac0ddf-20250801-075515-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)